### PR TITLE
Braces - (re)indenting comment issues

### LINKED ISSUE #3819: PSR2 (default) ruleset adds trailing whitespace
#### The PHP version you are using (`$ php -v`):
`7.0.27-0+deb9u1`

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
`PHP CS Fixer 2.12.0 Long Journey by Fabien Potencier and Dariusz Ruminski (a53f39a)`

#### The command you use to run PHP CS Fixer:
`php-cs-fixer fix myfile.php`

#### The configuration file you are using, if any:
N/A

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
<?php
class MyClass extends SomeClass {
/*	public function myFunction() {

		$MyItems = [];

		return $MyItems;
	}
*/
}
```
For clarity about how the whitespace starts out, here is the above snippet after passing through `cat -A`:
```<?php$
class MyClass extends SomeClass {$
/*^Ipublic function myFunction() {$
$
^I^I$MyItems = [];$
$
^I^Ireturn $MyItems;$
^I}$
*/$
}$
?>$

```

 * with unexpected changes applied when running PHP CS Fixer:
```php
<?php
class MyClass extends SomeClass
{
    /*	public function myFunction() {
    
            $MyItems = [];
    
            return $MyItems;
        }
    */
}
```

In case you can't see the added whitespace in the code sample above, here is what it now looks like after passing through `cat -A`:
```
<?php$
class MyClass extends SomeClass$
{$
    /*^Ipublic function myFunction() {$
    $
            $MyItems = [];$
    $
            return $MyItems;$
        }$
    */$
}$
```

 * with the changes you expected instead:
```php
<?php
class MyClass extends SomeClass
{
    /*	public function myFunction() {

            $MyItems = [];

            return $MyItems;
        }
    */
}
```

I.e. after passing through `cat -A`, the output from PHP-CS-Fixer should instead look like this:
```
<?php$
class MyClass extends SomeClass$
{$
    /*^Ipublic function myFunction() {$
$
            $MyItems = [];$
$
            return $MyItems;$
        }$
    */$
}$
```

### LINKED ISSUE #4814: [Bug] Trim trainling spaces not done in comment parts
Hello,

On the last version at this day, commented php part are not properly processed to trim trailing spaces.

Exemple:

```
        $this->view->query = $query;
        $this->view->listing = $listing;
        $this->view->ajax = $ajax;
        $this->view->paginator = $paginator;
        $this->view->areas_select = $areasSelect;
        $this->view->medias_select = $mediasSelect;
        $this->view->brands_select = $brandsSelect;
        $this->view->targets_select = $targetsSelect;
<= here OK
        /*
                // Search (query)
                $query = $request->get('query');
                <= here KO
                if (!empty($query))
                {
                    $news->addConditionParam('title LIKE ?', "%{$query}%");
                    $news->addConditionParam ('description LIKE ?', "%{$query}%", 'OR');
                    $filters[] = "query={$query}";
                }
                <= here KO
                // Pagination
                $filters[] = "page={$request->get('page')}";
                <= here KO
                // $filters_string = '?' . implode('&', $filters);
        */
```

I think for file coherence it should be done in this part too. Usualy when "trim trainling spaces" features are applied on a file, it is on all the file.
I should be applied in the case of mixed content too (html/php). I don't think it can break HTML code (HTML space are important because it can change behaviour of how diplayed, but as this is empty line, the empty line will act like a space for hmtl displaying). Expert could confirm ?

Thanks.

### PR DESCRIPTION
**Pull request by SpacePossum**
fixes #3819
fixes #4814

replaces https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4700

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
What's the status here?

**Comment by SpacePossum** [comment]
1,5 year old... I can't say anymore...  I really hope someone can have a look...

also the issue itself doesn't sound that hard;
when the `braces` fixer (re)indents a multi line comment it should not add indentation to empty lines in that comment

**Comment by SpacePossum** [comment]
this is good for review

File: src/Fixer/Basic/BracesFixer.php [hunk 1]
```diff
@@ -936,8 +936,8 @@ private function ensureWhitespaceAtIndexAndIndentMultilineComment(Tokens $tokens
             $tokens[$nextTokenIndex] = new Token([
                 $nextToken->getId(),
                 Preg::replace(
-                    '/(\R)'.$this->detectIndent($tokens, $nextTokenIndex).'/',
-                    '$1'.Preg::replace('/^.*\R([ \t]*)$/s', '$1', $whitespace),
+                    '/(\R)'.$this->detectIndent($tokens, $nextTokenIndex).'([\h]*\S+.*)/',
```
Comment (julienfalque) [inline src/Fixer/Basic/BracesFixer.php]: Wrapping `\h` inside `[]` is not needed here I think?

**Pull request merged**